src/urllib3/util/url.py

```diff
@@ -300,7 +300,7 @@ def _normalize_host(
 def _idna_encode(name: str) -> bytes:
     if name and any([ord(x) > 128 for x in name]):
         try:
-            import idna  # type: ignore
+            import idna
```

Comment:
Doesn't the latest idna include type hints?

Reply:
I think it does. but we need to install it in `idna` in the `lint` environment.

Reply:
Let's do that!

Reply:
Ok, I will create a new PR for this.